website/docs/script/sudt-script.mdx

```diff
@@ -0,0 +1,383 @@
+    }
+
+    0 // Success
+}
+```
+
+3. **Error Handling**: The script also includes robust error handling. It defines a custom `Error` enum to manage various error conditions, such as when the input amount is invalid or when there are issues with encoding. This helps ensure that any problems are clearly communicated.
+
+This implementation ensures that UDT transactions are validated correctly, maintaining the integrity of token transfers on the CKB blockchain. By checking both the owner mode and the amounts, the script helps prevent errors and ensures smooth operation.
+
+### Collecting UDT Amount
+
+In the sudt Script, the functions `collect_inputs_amount` and `collect_outputs_amount` play a crucial role in gathering the total amounts of UDTs from the respective input and output cells.
+
+Here’s how they work:
+
+Both functions iterate through the relevant cells and collect the `cell_data` to calculate the total UDT amount. This is done using the following syscalls:
+
+- **`Source::GroupInput`**: This ensures that only the cells with the same script as the current running script are iterating
+- **`Source::GroupOutput`**: Similarly, this syscall only the cells with the same script as the current running script are iterating
+
+By using these specific [sources](/docs/script/syscalls-for-script#source), the script avoids issues related to different UDT cell types, ensuring that only the appropriate cells are processed. This is particularly important in a blockchain environment where multiple UDTs may exist, and we want to ensure that the calculations are accurate and relevant to the current UDT Script.
+
+Here’s a brief look at how these functions are implemented:
+
+```rust
+fn collect_inputs_amount() -> Result<u128, Error> {
+    let mut buf = [0u8; UDT_AMOUNT_LEN];
+
+    let udt_list = QueryIter::new(load_cell_data, Source::GroupInput)
+        .map(|data| {
+            if data.len() >= UDT_AMOUNT_LEN {
+                buf.copy_from_slice(&data);
+                Ok(u128::from_le_bytes(buf))
+            } else {
+                Err(Error::AmountEncoding)
+            }
+        })
+        .collect::<Result<Vec<_>, Error>>()?;
+    Ok(udt_list.into_iter().sum::<u128>())
+}
+
+fn collect_outputs_amount() -> Result<u128, Error> {
+    let mut buf = [0u8; UDT_AMOUNT_LEN];
+
+    let udt_list = QueryIter::new(load_cell_data, Source::GroupOutput)
+        .map(|data| {
+            if data.len() >= UDT_AMOUNT_LEN {
+                buf.copy_from_slice(&data);
+                Ok(u128::from_le_bytes(buf))
+            } else {
+                Err(Error::AmountEncoding)
+            }
+        })
+        .collect::<Result<Vec<_>, Error>>()?;
+    Ok(udt_list.into_iter().sum::<u128>())
+}
+```
+
+In summary, these functions are essential for accurately calculating the total UDT amounts involved in the transaction while ensuring that only the relevant cells are considered, thus maintaining the integrity of the UDT transfer process.
```

Comment:
In summary, these functions are essential for the accurate calculation of the total UDT amounts involved in the transaction, while ensuring that only the relevant Cells are considered, thus maintaining the integrity of the UDT transfer process.